### PR TITLE
[Scheduling Sample] Appointment details page

### LINKED ISSUE #4930: [Scheduling Sample] Appointment details page (details & resources)
- /Appointment/:id
- Patient summary (left ⅓)
- Appointment details (middle ⅓, tab)
- Encounter history (tab, middle ⅓)
- Upcoming appointments (middle ⅓, tab)
- Past appointments (middle ⅓, tab)

### PR DESCRIPTION
**Pull request by filipeximenes**
# Description
Adds the Appointment detail page (/Appointment/:id), including patient profile, appointment details, list of encounters, and upcoming and past appointments for the same Patient.

## Page tabs
https://github.com/user-attachments/assets/5ed14aad-1105-4fcb-9c44-0f18f9849bca

## Complete, cancel and reschedule actions
https://github.com/user-attachments/assets/776a0179-cdf6-4b1a-affc-153acafc0458

## Create encounter action (edit: not opening a modal anymore)  
https://github.com/user-attachments/assets/52d4eca8-dfc1-40ff-92aa-b43d62f60f4a

Closes #4930 and #4931

### CODE REVIEW

File: examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      period: {
+        start: startDateTime,
+        end: endDateTime,
+      },
+      subject: createReference(patient),
+      participant: [
+        {
+          // Uses the logged user as the attender
+          individual: createReference(profile),
+          type: [{ coding: [{ system: 'http://terminology.hl7.org/CodeSystem/v3-ParticipationType', code: 'ATND' }] }],
```
Comment (filipeximenes) [inline examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx:55]: I wasn't sure if I should use ATND or PPRF

**Comment by filipeximenes** [comment]
Updated the UX, now clicking to create an encounter won't open a modal and it will use Appointment data to do it in the background.

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    try {
+      await medplum.createResource({
+        resourceType: 'Encounter',
+        status: 'finished',
+        subject: createReference(patient),
+        appointment: [createReference(appointment)],
+        class: {
+          system: 'http://terminology.hl7.org/CodeSystem/v3-ActCode',
+          code: 'AMB',
+          display: 'ambulatory',
```
Comment (rahul1) [inline examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:75]: For the purposes of the demo, let's set these to "Virtual"

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        },
+        serviceType: appointment.serviceType?.[0],
+        period: {
+          start: appointment.start,
+          end: appointment.end,
+        },
+        participant: [
+          {
```
Comment (rahul1) [inline examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:83]: We should make sure to include all the participants listed in the Appointment.participant as participants in the encounter

**Review by rahul1 — COMMENTED** [review]
A few minor comments, but then good to merege

File: examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx (hunk 1 above)
Comment (rahul1) [reply examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx]: ATND is fine

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+          system: 'http://terminology.hl7.org/CodeSystem/v3-ActCode',
+          code: 'AMB',
+          display: 'ambulatory',
+        },
+        serviceType: appointment.serviceType?.[0],
+        period: {
+          start: appointment.start,
+          end: appointment.end,
+        },
+        participant: [
```
Comment (rahul1) [inline examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:82]: nit: can you also populate `Encounter.length`. While this is easily calculable from start/end, it's a key field needed for billing and is also searchable.

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx (hunk 3 above)
Comment (filipeximenes) [reply examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:83]: For `Encounter` there's a `subject` field to store the patient and `participant` is according to the docs `The list of people responsible for providing the service.` so just to confirm, I should filter out the Patient from the `participant` and add everyone else, right?

**Review by rahul1 — APPROVED** [review]

**Pull request merged**